# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/QuicSocketControl.h
# Commit: 0bf045553a40
# Full Hash: 0bf045553a4079b9948e1814e5cd1038be301b0b
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2019-11-02 09:54:21
# Regressor Bug: 1581637
# File Overlap Count: 1
# Description:
#   Bug 1581637 - Part 8 - Add Http3Session/Http3Stream. r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46652
# ==============================================================================

diff -r 2801613e5d2d -r 0bf045553a40 netwerk/protocol/http/QuicSocketControl.h
--- a/netwerk/protocol/http/QuicSocketControl.h	Fri Nov 01 14:55:56 2019 +0000
+++ b/netwerk/protocol/http/QuicSocketControl.h	Fri Nov 01 14:55:55 2019 +0000
@@ -12,8 +12,7 @@
 namespace mozilla {
 namespace net {
 
-// Will be added when Http3 lands.
-// class Http3Session;
+class Http3Session;
 
 // IID for the QuicSocketControl interface
 #define NS_QUICSOCKETCONTROL_IID                     \
@@ -37,8 +36,7 @@
   void SetInfo(uint16_t aCipherSuite, uint16_t aProtocolVersion,
                uint16_t aKeaGroup, uint16_t aSignatureScheme);
 
-  // Will be added when Http3 lands.
-  // void SetAuthenticationCallback(Http3Session *aHttp3Session);
+  void SetAuthenticationCallback(Http3Session* aHttp3Session);
   void CallAuthenticated();
 
   void HandshakeCompleted();